Arguments

juanmbellini opened the pull request:
**WARNING: DO NOT MERGE TILL #2 IS MERGED**
**WARNING: ONCE #2 IS MERGED, REBASE WITH DEV**

juanmbellini committed:
Add program arguments

juanmbellini committed:
Wire ProgramArguments with main class

- Refactor the ParticleProvider class

juanmbellini commented on src/main/java/ar/edu/itba/ss/hourglass/utils/ParticleProvider.java:
```diff
@@ -3,53 +3,95 @@
+            final double radius = minRadius + new Random().nextDouble() * (maxRadius - minRadius);
+            if (particles.stream().noneMatch(p -> p.doOverlap(position, radius))) {
+                particles.add(new Particle(this.mass, radius, position, Vector2D.ZERO, Vector2D.ZERO));
+                addMore = canAddMore(particles); // Check if more particles can be added.
```
Por ahi podemos poner como condicion de corte cuando no entró en el if N veces seguidas (o sea, trato de meter una particula durante N veces y siempre overlappeo)

matifraga replied:
Podemos hacer eso y usarlo como indicador de la cantidad de particulas que soportamos como máxima en el silo.

juanmbellini replied:
Claro claro.

juanmbellini committed:
Remove unused properties

juanmbellini committed:
Add the Kn argument to the Silo

matifraga commented on src/main/java/ar/edu/itba/ss/hourglass/models/Silo.java:
```diff
@@ -38,31 +40,52 @@
+        this.topWall = Wall.getHorizontal(new Vector2D(length, 0), width);
+        this.normalElasticConstant = normalElasticConstant;
+        this.particleProvider = new ParticleProvider(minRadius, maxRadius, mass,
+                leftBottomWall.getInitialPoint().getX(), rightWall.getFinalPoint().getX(),
```
porqué leftBottomWall y no leftWall ?

juanmbellini replied:
Es lo mismo en realidad. Ambas tienen el mismo initial point

matifraga commented on src/main/java/ar/edu/itba/ss/hourglass/models/Silo.java:
```diff
@@ -38,31 +40,52 @@
+        this.normalElasticConstant = normalElasticConstant;
+        this.particleProvider = new ParticleProvider(minRadius, maxRadius, mass,
+                leftBottomWall.getInitialPoint().getX(), rightWall.getFinalPoint().getX(),
+                leftBottomWall.getInitialPoint().getY(), rightWall.getFinalPoint().getY());
```
No sería initialPoint en rightWall? para que estén arriba de todo

juanmbellini replied:
nono, las partículas tienen que crearse en todo el silo.
Igualmente estos argumentos indican el xMin - xMax e yMin - yMax para los randoms.

matifraga replied:
Claro, pero de esa manera como reinsertas particulas? tendriamos otro metodo para reinsertarlas?

matifraga commented on src/main/java/ar/edu/itba/ss/hourglass/utils/ParticleProvider.java:
```diff
@@ -3,53 +3,95 @@
+        this.xMin = xMin;
+        this.xMax = xMax;
+        this.yMin = yMin;
+        this.yMax = yMax;
```
Esto deberíamos sacarlo teniendo en cuenta el doc, deberíamos ir insertando partículas de abajo hacia arriba en el silo (a medida que se vaya llenando, por ejemplo podemos aumentar un poco la posición, 2 diametros, si pasan N overlaps consecutivos) y tener guardada siempre la altura de la partícula que está mas arriba para poder reinsertar partículas cuando salen del silo)

matifraga replied:
Lo que tenemos que lograr con esto, y el motivo por el que tiene que ser así, es porque no podemos aumentar mucho la energía cinética de todo el sistema, porque explota por los aires, al insertar partículas siempre cerca de otras, la energía cinética que introducís al sistema es baja y no genera tantas disrupciones.

juanmbellini committed:
Fix the createParticles method

- Use as stop condition the amount of consecutive failed tries
- Also fix bug in Particle#doOverlap method, where the comparison was being made the other way round.
